pkg/module/file/README.md

```diff
@@ -0,0 +1,183 @@
+# File Module
+
+The `file` module provides file transfer capabilities between the client and dutagent, allowing you to upload files to the device and download files from the device.
+
+
+## Usage
+
+### Command Syntax
+
+The file module supports two argument forms:
+
+```bash
+# Single path
+dutctl device cmd <path>
+
+# Colon syntax
+dutctl device cmd <source>:<destination>
+```
+
+The operation type (upload or download) must be configured in the device YAML.
+
+### Destination Path Resolution Priority
+
+The module determines the final source and destination paths using this priority order:
+
+1. **`default_destination` in config** - If set, always takes precedence
+2. **Colon syntax** - If provided (`:` separator), overrides defaults
+3. **Working directory + basename** - Fallback if neither above is set
+
+**Examples:**
+
+```bash
+# Uses default_destination from config
+dutctl device1 upload-firmware ./firmware.bin
+
+# Colon syntax - only works if default_destination is NOT set
+dutctl device1 upload-firmware ./firmware.bin:/custom/path/firmware.bin
+
+# If no default_destination, uploads to working-dir/firmware.bin
+dutctl device1 upload-firmware ./firmware.bin
+```
+
+## Configuration Options
+
+Options are configured in the device YAML configuration:
+
+| Option | Type | Default | Description |
+|--------|------|---------|-------------|
+| `operation` | string | **required** | Operation type: `"upload"` or `"download"` |
+| `default_destination` | string | optional | **Takes highest priority.** Destination path on either dutagent or dut-client. If set, colon syntax is ignored. |
+| `forcedir` | bool | `false` | Force creation of parent directories if they don't exist (upload only) |
+| `overwrite` | bool | `false` | Overwrite existing destination files (upload only) |
+| `mode` | string | `""` | File permissions in octal format, e.g., `"0644"`, `"0755"` (upload only) |
+
```

Comment:
Maybe we could also add `source` in the same manner and the same mechanics as `destination`.

Reply:
So i have been thinking about this a lot and i am not sure. So the actual most common use case for downloading to the client is setting the source path fixed, for uploading it is setting the destination path fixed. 
Setting the source fixed for upload does not seem right to me. 
What do you think if we add a config option named path, which sets either source/destination path depending on mode. The only backdraw is that it might be confusing. So another option would be to allow destination only on upload, and source on download in the config.